ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/PeerSimulator/Run.hs

```diff
@@ -272,7 +274,9 @@ runPointSchedule schedulerConfig genesisTest tracer0 =
           = pure nullTracer
     stateTracer <- mkStateTracer
     BlockFetch.startBlockFetchLogic registry chainDb fetchClientRegistry getCandidates
-    void $ forkLinkedThread registry "ChainSel trigger" (reprocessLoEBlocksOnCandidateChange chainDb getCandidates)
+    -- void $ forkLinkedThread registry "ChainSel trigger" (reprocessLoEBlocksOnCandidateChange chainDb getCandidates)
+    void $ forkLinkedThread registry "LoE updater background" $
+      runLoEUpdaterBackground chainDb getLatestSlots
```

Comment:
Should the LoE fragment continue to be updated when `getCandidates` changes? I don't see now where this happens.

Reply:
Yes, when `getCandidates` changes, `getLatestSlots` also changes, which is used as the trigger above.

Reply:
The latest slot is set in a separate transaction from the candidate fragment and "their" tip. It is possible that the GDD runs before the candidate is updated. Either everything needs to be done on a single transaction now, or this PR needs to take other measures to update after candidates change.

Reply:
Also it looks like the last slot isn't updated at all in `nextStep`?